crates/ruff_server/src/edit/text_document.rs

```diff
@@ -115,15 +114,11 @@ impl TextDocument {
                 new_contents = change;
             }
 
-            if new_contents != old_contents {
-                active_index = LineIndex::from_source_text(&new_contents);
-            }
+            active_index = LineIndex::from_source_text(&new_contents);
         }
 
         self.modify_with_manual_index(|contents, version, index| {
-            if contents != &new_contents {
-                *index = active_index;
-            }
```

Comment:
I don't think it's worth comparing the line index here just to avoid a pointer mutation.